File: app/src/main/java/com/timotiusoktorio/pencake/ui/dialogs/AlertDialogFragment.java
package com.timotiusoktorio.pencake.ui.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;

public class AlertDialogFragment extends DialogFragment {

    private static final String ARG_DIALOG_TITLE = "ARG_DIALOG_TITLE";
    private static final String ARG_DIALOG_MESSAGE = "ARG_DIALOG_MESSAGE";

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        String title = getArguments().getString(ARG_DIALOG_TITLE);
        String message = getArguments().getString(ARG_DIALOG_MESSAGE);
        return new AlertDialog.Builder(getActivity())
                .setTitle(title)
                .setMessage(message)
                .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dismiss();
                    }
                })
                .create();
    }

    public static AlertDialogFragment newInstance(String title, String message) {
        Bundle args = new Bundle();
        args.putString(ARG_DIALOG_TITLE, title);
        args.putString(ARG_DIALOG_MESSAGE, message);
        AlertDialogFragment fragment = new AlertDialogFragment();
        fragment.setArguments(args);
        return fragment;
    }
}

File: app/src/main/java/com/timotiusoktorio/pencake/ui/dialogs/AlertDialogFragment.kt
package com.timotiusoktorio.pencake.ui.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.os.Bundle
import org.jetbrains.anko.bundleOf

class AlertDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val title = arguments!!.getString(ARG_DIALOG_TITLE)
        val message = arguments!!.getString(ARG_DIALOG_MESSAGE)
        return AlertDialog.Builder(activity)
                .setTitle(title)
                .setMessage(message)
                .setPositiveButton(android.R.string.ok) { _, _ -> dismiss() }
                .create()
    }

    companion object {

        private const val ARG_DIALOG_TITLE = "ARG_DIALOG_TITLE"
        private const val ARG_DIALOG_MESSAGE = "ARG_DIALOG_MESSAGE"

        fun newInstance(title: String, message: String): AlertDialogFragment {
            val args = bundleOf(ARG_DIALOG_TITLE to title, ARG_DIALOG_MESSAGE to message)
            val fragment = AlertDialogFragment()
            fragment.arguments = args
            return fragment
        }
    }
}